[The CI job failed](https://github.com/tauri-apps/wry/actions/runs/7006878921/job/19059798479?pr=1099) correctly since we haven't merged #1098 yet.

```
error[E0636]: the feature `doc_cfg` has already been declared
   --> src/lib.rs:172:29
    |
172 | #![cfg_attr(docsrs, feature(doc_cfg))]
    |                             ^^^^^^^
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -1,6 +1,10 @@
 name: test
 
-on: [pull_request]
+on:
+  push:
+    branches:
+      - dev
+  pull_request:
 
 jobs:
   test:
@@ -17,7 +21,7 @@ jobs:
     runs-on: ${{ matrix.platform.os }}
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
       - name: install stable
         uses: dtolnay/rust-toolchain@stable
         with:
@@ -59,3 +63,23 @@ jobs:
       - name: Run tests with miri
         if: (!contains(matrix.platform.target, 'android') && !contains(matrix.platform.target, 'ios'))
         run: cargo +nightly miri test --verbose --target ${{ matrix.platform.target }} --features linux-body
+
+  doc:
+    runs-on: ubuntu-latest
+    env:
+      RUSTFLAGS: '--cfg docsrs'
+      RUSTDOCFLAGS: '--cfg docsrs'
+    steps:
+      - uses: actions/checkout@v4
+      - name: install nightly
+        uses: dtolnay/rust-toolchain@nightly
+        with:
+          targets: x86_64-unknown-linux-gnu,x86_64-pc-windows-msvc,x86_64-apple-darwin
+      - name: install webkit2gtk
+        run: |
+          sudo apt-get update -y -q
+          sudo apt-get install -y --no-install-recommends libwebkit2gtk-4.1-dev
+      # When updating arguments to `cargo doc`, consider to update `[package.metadata.docs.rs]` section in Cargo.toml
+      - run: cargo doc -p wry --no-default-features --features=file-drop,protocol,os-webview --target=x86_64-unknown-linux-gnu
+      - run: cargo doc -p wry --no-default-features --features=file-drop,protocol,os-webview --target=x86_64-pc-windows-msvc
+      - run: cargo doc -p wry --no-default-features --features=file-drop,protocol,os-webview --target=x86_64-apple-darwin

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -66,20 +66,29 @@ jobs:
 
   doc:
     runs-on: ubuntu-latest
-    env:
-      RUSTFLAGS: '--cfg docsrs'
-      RUSTDOCFLAGS: '--cfg docsrs'
     steps:
       - uses: actions/checkout@v4
       - name: install nightly
         uses: dtolnay/rust-toolchain@nightly
-        with:
-          targets: x86_64-unknown-linux-gnu,x86_64-pc-windows-msvc,x86_64-apple-darwin
       - name: install webkit2gtk
         run: |
           sudo apt-get update -y -q
           sudo apt-get install -y --no-install-recommends libwebkit2gtk-4.1-dev
-      # When updating arguments to `cargo doc`, consider to update `[package.metadata.docs.rs]` section in Cargo.toml
-      - run: cargo doc -p wry --no-default-features --features=file-drop,protocol,os-webview --target=x86_64-unknown-linux-gnu
-      - run: cargo doc -p wry --no-default-features --features=file-drop,protocol,os-webview --target=x86_64-pc-windows-msvc
-      - run: cargo doc -p wry --no-default-features --features=file-drop,protocol,os-webview --target=x86_64-apple-darwin
+      - name: Install dasel
+        run: |
+          curl -LO 'https://github.com/TomWright/dasel/releases/latest/download/dasel_linux_amd64.gz'
+          gunzip dasel_linux_amd64.gz
+          chmod +x dasel_linux_amd64
+          mv dasel_linux_amd64 dasel
+          ./dasel --version
+      - name: Run cargo doc with each targets
+        run: |
+          set -ex
+          export RUSTFLAGS="$(./dasel -r toml -w - 'package.metadata.docs.rs.rustc-args.all().join( )' < Cargo.toml)"
+          export RUSTDOCFLAGS="$(./dasel -r toml -w - 'package.metadata.docs.rs.rustdoc-args.all().join( )' < Cargo.toml)"
+          features="$(./dasel -r toml -w - 'package.metadata.docs.rs.features.all().join(\,)' < Cargo.toml)"
+
+          for target in $(./dasel -r toml -w - 'package.metadata.docs.rs.targets.all().join( )' < Cargo.toml); do
+            rustup target add "$target"
+            cargo doc -p wry --no-default-features "--features=$features" "--target=$target"
+          done

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -74,21 +74,19 @@ jobs:
         run: |
           sudo apt-get update -y -q
           sudo apt-get install -y --no-install-recommends libwebkit2gtk-4.1-dev
-      - name: Install dasel
-        run: |
-          curl -LO 'https://github.com/TomWright/dasel/releases/latest/download/dasel_linux_amd64.gz'
-          gunzip dasel_linux_amd64.gz
-          chmod +x dasel_linux_amd64
-          mv dasel_linux_amd64 dasel
-          ./dasel --version
       - name: Run cargo doc with each targets
         run: |
           set -ex
-          export RUSTFLAGS="$(./dasel -r toml -w - 'package.metadata.docs.rs.rustc-args.all().join( )' < Cargo.toml)"
-          export RUSTDOCFLAGS="$(./dasel -r toml -w - 'package.metadata.docs.rs.rustdoc-args.all().join( )' < Cargo.toml)"
-          features="$(./dasel -r toml -w - 'package.metadata.docs.rs.features.all().join(\,)' < Cargo.toml)"
 
-          for target in $(./dasel -r toml -w - 'package.metadata.docs.rs.targets.all().join( )' < Cargo.toml); do
+          md="$(cargo metadata --format-version=1 | jq '.packages[] | select(.name=="wry") | .metadata.docs.rs')"
+
+          export RUSTDOCFLAGS="$(echo "$md" | jq -r '.["rustdoc-args"] | join(" ")')"
+          export RUSTFLAGS="$(echo "$md" | jq -r '.["rustc-args"] | join(" ")')"
+
+          features="$(echo "$md" | jq -r '.features | join(",")')"
+
+          for target in $(echo "$md" | jq -r '.targets | join(" ")')
+          do
             rustup target add "$target"
             cargo doc -p wry --no-default-features "--features=$features" "--target=$target"
           done

